cmd/maya-exporter/app/collector/collector.go

```diff
@@ -122,6 +139,16 @@ func (c *collector) Collect(ch chan<- prometheus.Metric) {
 		stats       stats
 	)
 
+	if c.isrequestInFlightInProgress() {
+		c.targetRejectRequestCounter.Inc()
+		c.targetRejectRequestCounter.Collect(ch)
+		return
+	}
+
+	c.Lock()
```

Comment:
Can these three lines be in a function -- just like for setting false? 
What is the reason for following different approaches for setting false vs setting to true?

Reply:
Done